core/src/robot/BasicTransmissions.cc

```diff
@@ -0,0 +1,47 @@
+#include <algorithm>
+
+#include "jiminy/core/utilities/Helpers.h"
+
+#include "jiminy/core/robot/BasicTransmissions.h"
+
+
+namespace jiminy
+{
+    SimpleTransmission::SimpleTransmission(std::string const & name) :
+    AbstractTransmissionBase(name)
+    {
+        std::cout << "construct simple transmission" << std::endl;
+    }
+
+    hresult_t SimpleTransmission::setOptions(configHolder_t const & /*transmissionOptions*/)
+    {
+        return hresult_t::SUCCESS;
+    }
+
+    void SimpleTransmission::computeJacobian(vectorN_t const & /*q*/,
+                                             matrixN_t & /*Jac*/)
+    {
+        // TODO
+    }
+
+    void SimpleTransmission::computeInverseJacobian(vectorN_t const & /*q*/,
+                                                    matrixN_t & /*Jac*/)
+    {
+        // TODO
+    }
+
+    void SimpleTransmission::computeTransform(vectorN_t const & /*qMotors*/,
+                                              vectorN_t       & /*qJoints*/)
```

Comment:
`std::vector<vectorN_t> & qJoints` for convenience.

Reply:
This method is not straightforward to implement. You need to do something different depending on the type of joints. So you need to loop over `jointTypes` and `qJoints` jointly using iterators, check want do to based on the type, and keep track of the `motorPositionIdx` that you must update manually accordingly.